### ISSUE TITLE
Italic lowercase `f` looks like +U0192

### ISSUE THREAD

**Issue by gwk** [JetBrains/JetBrainsMono#129]
This is a purely stylistic opinion, but I figure it can't hurt to share (forgive me if I misuse the terminology; I'm no expert!)

The italic `f` has a curved descender below the baseline. To my eye, this makes it look unlike the rest of the font, and several times I have thought it was the unicode character "ƒ" (+U0192, LATIN SMALL LETTER F WITH HOOK). This is especially distracting because in languages like javascript and swift, "function" or "func" is an italicized keyword and so I see it all the time.

Looking at the font in macOS Font Book and toggling between regular and italic, it's obvious that `f` and `y` are the only characters with geometric differences like this.

I suggest removing the descender  on italic `f`. The `y` seems fine, but perhaps you will want to reconsider it as well.

Thanks again for open sourcing the font, it's very cool.

**Comment by AlsoScratch** [comment]
We could have an 'oblique' variant, that's one where the glyphs are leaning to the right without any cursive-ness.

**Comment by philippnurullin** [comment]
@gwk The problem you trying to solve is that italic `f` is mistaken by `ƒ`. Do you have example when `ƒ` is used in code? 

@AlsoScratch It can be achieved by adding the alternate style sets, which is not top priority right now, so it will take some time.

**Comment by xxkfqz** [comment]
> @gwk The problem you trying to solve is that italic `f` is mistaken by `ƒ`. Do you have example when `ƒ` is used in code?

I don't know where one can find `ƒ` in code, but it looks really similar
![f](https://user-images.githubusercontent.com/52203435/77747150-f2fae180-703f-11ea-8164-d65aa3680c25.png)

**Comment by gwk** [comment]
I think there are two prongs to this issue.

One is the question that @philippnurullin poses. I do not have an example on hand where +U0192 is used in code. I do however look at lots of unicode text in the editor (for example, web scraping projects give me lots of dirty text that needs to be cleaned up). I also sometimes accidentally insert weird characters into my code, for example if I hold down option instead of control for a keyboard command. The bottom line for me personally is that I try to pay close attention to weird glyphs in text, and this italic 'f' sticks out, making me wonder if it is really an 'f'.

I hesitate to argue the finer points of this angle. For example, you could take the stance that it doesn't really matter because it only shows up when the syntax highlighter italicizes a keyword. Broadly speaking, it's not a big deal.

In my mind, a more productive framing is to ask if this particular design element (the 'f' descender) adds anything positive. I suspect the descender was added as a creative flourish, or perhaps a nod to typographic conventions for italics (and  perhaps there are other principles that I am unaware of). I respect that those are real points of value, however small or intangible. I'm simply suggesting that the flourish also incurs small but real costs.

Ultimately, this is meant as constructive feedback to the font designer, who might find it worthwhile to reconsider the cost/benefit of this small detail.

If I had the software on hand, I would make a variant without the descender myself, just for the experience!

**Comment by philippnurullin** [comment]
You sound like the type designer is somewhere els. You know it's me, right? )

The reading process is not based on continuous comparison of letters. We always having context of the letter sequence in a word, so the eye is not even registering every letter. The context is making this symbol unmistakable. Basically you know when you using the `f` & when you using the `ƒ` because the usage of the second is really limited & specific.

The italics as right now had to have this construction of the `f`. If not it will fallout from the system & will draw attention. The descender element is rhymed with similar elements in letters like `y` `j` `g`.

The best solution for your case will be having the oblique italics, which is achievable by stylistic set. Can't say the date to this feature sadly.

**Comment by gwk** [comment]
@philippnurullin my apologies. After watching the no-ligatures issue go by, I think I was assuming this would turn into bike-shedding with strangers. I appreciate your explanation, expertise, and judgment.

The crux of my plea is that I don't trust context as much as you assume when reading text files. I fully concede that in the common case you are right. But I've encountered enough weird characters where they shouldn't have been, that wasted tons of my time, that I'm sensitive to this. This is really the only reason that I've opted for the no-ligature version. I like the ligatures! They are cool and well designed. But they add ambiguity and thus cognitive overhead that slows me down.

The specific problem I have with the current 'f' design is that because it has both ascender and descender, it looks taller than I think any letter should be, so I assume it is an "alien".

Just for fun, I tried out two alternatives:
<img width="495" alt="fay-fay-fay" src="https://user-images.githubusercontent.com/406380/77924765-f4532680-7258-11ea-87d5-1a1eaf7ed280.png">

In any case, thanks for considering this and taking the time to discuss. I think you have done an excellent job with this project and I'm grateful that JetBrains open sourced it!
